common/scripts/audit.js

```diff
@@ -0,0 +1,145 @@
+ *--------------------------------------------------------------------------------------------*/
+
+const fs = require("fs");
+const path = require("path");
+const { spawn } = require("child_process");
+const {
+  logBuildError,
+  logBuildWarning,
+  failBuild,
+  throwAfterTimeout,
+} = require("./utils");
+
+const rushCommonDir = path.join(__dirname, "../");
+
+(async () => {
+  const commonTempDir = path.join(rushCommonDir, "config/rush");
+
+  // Npm audit will occasionally take minutes to respond - we believe this is just the npm registry being terrible and slow.
+  // We don't want this to slow down our builds though - we'd rather fail fast and try again later.  So we'll just timeout after 30 seconds.
+  let jsonOut = {};
+  try {
+    console.time("Audit time");
+    jsonOut = await Promise.race([
+      runYarnAuditAsync(commonTempDir),
+      throwAfterTimeout(180000, "Timed out contacting npm registry."),
+    ]);
+    console.timeEnd("Audit time");
+    console.log();
+  } catch (error) {
+    // We want to stop failing the build on transient failures and instead fail only on high/critical vulnerabilities.
+    logBuildWarning(error);
+    process.exit();
+  }
+
+  if (jsonOut.error) {
+    console.error(jsonOut.error.summary);
+    logBuildWarning(
+      "Rush audit failed. This may be caused by a problem with the npm audit server."
+    );
+  }
+
+  // A list of temporary advisories excluded from the High and Critical list.
+  // Warning this should only be used as a temporary measure to avoid build failures
+  // for development dependencies only.
+  // All security issues should be addressed asap.
+  const excludedAdvisories = [
+    1754, // https://npmjs.com/advisories/1754
+  ];
+
+  let shouldFailBuild = false;
+  for (const action of jsonOut.actions) {
+    for (const issue of action.resolves) {
+      const advisory = jsonOut.advisories[issue.id];
+
+      const mpath = issue.path.replace("@rush-temp", "@bentley");
+
+      const severity = advisory.severity.toUpperCase();
+      const message = `${severity} Security Vulnerability: ${advisory.title} in ${advisory.module_name} (from ${mpath}).  See ${advisory.url} for more info.`;
+
+      // For now, we'll only treat CRITICAL and HIGH vulnerabilities as errors in CI builds.
+      if (
+        !excludedAdvisories.includes(advisory.id) &&
+        (severity === "HIGH" || severity === "CRITICAL")
+      ) {
+        logBuildError(message);
+        shouldFailBuild = true;
+      } else if (
+        excludedAdvisories.includes(advisory.id) ||
+        severity === "MODERATE"
+      )
+        // Only warn on MODERATE severity items
+        logBuildWarning(message);
+    }
+  }
+
+  // For some reason yarn audit can return the json without the vulnerabilities
+  if (undefined === jsonOut.metadata.vulnerabilities || shouldFailBuild)
+    failBuild();
+
+  process.exit();
+})();
+
+function runYarnAuditAsync(cwd) {
```

Comment:
doesn't this add a dep on yarn? will this work if you uninstall yarn globally

Reply:
switched to using the pnpm  one now that we got that other pr through, audit looks good now.